caravel/assets/visualizations/maps/ScatterPlotClusterOverlay.jsx

```diff
@@ -0,0 +1,230 @@
+      ctx.font = scale * maxWidth + 'px sans-serif';
+    }
+
+    ctx.fillText(label, pixel[0], pixel[1]);
+    ctx.globalCompositeOperation = this.props.compositeOperation;
+    ctx.shadowBlur = 0;
+    ctx.shadowColor = '';
+  }
+  /* eslint-enable no-param-reassign */
+
+  drawPoints() {
+    const props = this.props;
+    const pixelRatio = window.devicePixelRatio || 1;
+    const canvas = this.refs.overlay;
+    const ctx = canvas.getContext('2d');
+    const radius = props.pointRadius;
+    const mercator = ViewportMercator(props);
+    const rgb = props.rgb;
+    let maxLabel = -1;
+    const clusterLabelMap = [];
+
+    props.locations.forEach(function (location, i) {
+      if (location.get('properties').get('cluster')) {
+        let clusterLabel = location.get('properties').get('metric')
+          ? location.get('properties').get('metric')
+          : location.get('properties').get('point_count');
+
+        if (clusterLabel instanceof Immutable.List) {
+          clusterLabel = clusterLabel.toArray();
+          if (props.aggregatorName === 'mean') {
+            clusterLabel = d3.mean(clusterLabel);
+          } else if (props.aggregatorName === 'median') {
+            clusterLabel = d3.median(clusterLabel);
+          } else if (props.aggregatorName === 'stdev') {
+            clusterLabel = d3.deviation(clusterLabel);
+          } else {
+            clusterLabel = d3.variance(clusterLabel);
+          }
+        }
+
+        clusterLabel = isNumeric(clusterLabel)
+          ? d3.round(clusterLabel, 2)
+          : location.get('properties').get('point_count');
+        maxLabel = Math.max(clusterLabel, maxLabel);
+        clusterLabelMap[i] = clusterLabel;
+      }
+    }, this);
+
+    ctx.save();
+    ctx.scale(pixelRatio, pixelRatio);
+    ctx.clearRect(0, 0, props.width, props.height);
+    ctx.globalCompositeOperation = props.compositeOperation;
+
+    if ((props.renderWhileDragging || !props.isDragging) && props.locations) {
+      props.locations.forEach(function _forEach(location, i) {
+        const pixel = mercator.project(props.lngLatAccessor(location));
+        const pixelRounded = [d3.round(pixel[0], 1), d3.round(pixel[1], 1)];
+
+        if (pixelRounded[0] + radius >= 0
+          && pixelRounded[0] - radius < props.width
+          && pixelRounded[1] + radius >= 0
+          && pixelRounded[1] - radius < props.height) {
+          ctx.beginPath();
+          if (location.get('properties').get('cluster')) {
+            let clusterLabel = clusterLabelMap[i];
+            const scaledRadius = d3.round(Math.pow(clusterLabel / maxLabel, 0.5) * radius, 1);
+            const fontHeight = d3.round(scaledRadius * 0.5, 1);
+            const gradient = ctx.createRadialGradient(
+              pixelRounded[0], pixelRounded[1], scaledRadius,
+              pixelRounded[0], pixelRounded[1], 0
+            );
+
+            gradient.addColorStop(1, 'rgba(' + rgb[1] + ', ' + rgb[2] + ', ' + rgb[3] + ', 0.8)');
```

Comment:
nit:, the ES6 backtick formatting syntax is neat for `' + '`-heavy strings, as in
```
`rgba(${rgb[1]}, ${rgb[2]}, ${rgb[3]}, 0.8)`
```